### PR TITLE
Adding System.Drawing.Common.Drawing2D PathGradientBrush tests. #20711

### PR DESCRIPTION
**Pull request by KostaVlev**
Converting mono **PathGradientBrush** tests and adding few new. #20711

### CODE REVIEW

**Comment by hughbe** [comment]
Looks great!
I may be missing it but are there any tests for dispose here? It'd be good to make sure that ArgumentExceptions are thrown when invoking the native methods after disposal.
This is because some implementations (e.g mono's libgdiplus) may be missing argument validation for the "this" pointer and cause access violations

**Comment by KostaVlev** [comment]
I will add test for Dispose.

File: src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs [hunk 1]
```diff
@@ -0,0 +1,919 @@
...
+namespace System.Drawing.Drawing2D.Tests
+{
+    public class PathGradientBrushTests
+    {
+        private readonly Point[] _defaultIntPoints = new Point[2] { new Point(1, 2), new Point(20, 30) };
+        private readonly PointF[] _defaultFloatPoints = new PointF[2] { new PointF(1, 2), new PointF(20, 30) };
+        private readonly RectangleF _defaultRectangle = new RectangleF(1, 2, 19, 28);
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_Points_ReturmsExpected()
```
Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: type: ~Returms~ Returns

File: src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs [hunk 2]
```diff
@@ -0,0 +1,919 @@
...
+            yield return new object[] { WrapMode.Clamp };
+            yield return new object[] { WrapMode.Tile };
+            yield return new object[] { WrapMode.TileFlipX };
+            yield return new object[] { WrapMode.TileFlipXY };
+            yield return new object[] { WrapMode.TileFlipY };
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(WrapMode_TestData))]
+        public void Ctor_PointsWrapMode_ReturmsExpected(WrapMode wrapMode)
```
Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: type: ~Returms~ Returns

File: src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs [hunk 3]
```diff
@@ -0,0 +1,919 @@
...
+        public void Ctor_PointsNull_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("points", () => new PathGradientBrush((Point[])null));
+            AssertExtensions.Throws<ArgumentNullException>("points", () => new PathGradientBrush((PointF[])null));
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(0)]
+        [InlineData(1)]
+        public void Ctor_PointsLenghtLessThenTwo_ThrowsOutOfMemoryException(int pointsLeght)
```
Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: Typo: ~Lenght~ Length

Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: Also ~pointsLeght~ pointsLength

File: src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs [hunk 4]
```diff
@@ -0,0 +1,919 @@
...
+                Assert.Throws<NullReferenceException>(() => brush.Blend = new Blend() { Factors = null, Positions = new float[0] });
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Blend_NullBlendProperites_ThrowsArgumentNullException()
+        {
+            using (PathGradientBrush brush = new PathGradientBrush(_defaultFloatPoints))
+            {
+                AssertExtensions.Throws<ArgumentNullException>("source",() => brush.Blend = new Blend() { Factors = new float[0], Positions = null });
```
Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: nit: space after `"source",`

File: src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs [hunk 5]
```diff
@@ -0,0 +1,919 @@
...
+                Assert.Throws<NullReferenceException>(() => brush.InterpolationColors = null);
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void InterpolationColors_NullColors_ThrowsNullReferenceException()
+        {
+            using (PathGradientBrush brush = new PathGradientBrush(_defaultFloatPoints))
+            {
+                Assert.Throws<NullReferenceException>(() => brush.InterpolationColors = new ColorBlend() { Colors = null, Positions = null });
```
Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: nit: Some of these lines are a bit long. Can you put some of these onto multiple lines, if they are over ~120 characters?

Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: It's mainly the calls to Assert.Throws that get pretty long here.

File: src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs [hunk 6]
```diff
@@ -82,7 +82,7 @@ public void Ctor_PointsNull_ThrowsArgumentNullException()
         [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
         [InlineData(0)]
         [InlineData(1)]
-        public void Ctor_PointsLenghtLessThenTwo_ThrowsOutOfMemoryException(int pointsLeght)
+        public void Ctor_PointsLengthLessThenTwo_ThrowsOutOfMemoryException(int pointsLeght)
```
Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: The parameter here is still misspelled: "pointsLeght".

Comment (KostaVlev) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: Ughh... done :)

File: src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs [hunk 7]
```diff
@@ -82,12 +82,12 @@ public void Ctor_PointsNull_ThrowsArgumentNullException()
         [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
         [InlineData(0)]
         [InlineData(1)]
-        public void Ctor_PointsLengthLessThenTwo_ThrowsOutOfMemoryException(int pointsLeght)
+        public void Ctor_PointsLengthLessThenTwo_ThrowsOutOfMemoryException(int pointsLegth)
```
Comment (mellinoe) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: Not trying to be picky... but it's still misspelled 😄

Comment (KostaVlev) [inline src/System.Drawing.Common/tests/Drawing2D/PathGradientBrushTests.cs]: ha ha omg :))) I am fixing it

**Comment by mellinoe** [comment]
LGTM. Thanks!

**Pull request merged**